FuelSystem.cs

```diff
@@ -0,0 +1,55 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class FuelSystem : MonoBehaviour
+{
+    private MovementInputSystem movementInputSystem;
+
+    [SerializeField] private float maxFuelCapacity = 100f;
+    private float currentFuelLevel;
+    public Text fuelText;
+    public Color lowFuelColor = Color.red;
+    private float lowFuelThreshold = 0.33f;
```

Comment:
The threshold should be set in the inspector, not defined as a private variable in code